### PR TITLE
Contour 0.4 draft release notes

### PR DESCRIPTION
**Pull request by davecheney**
For March 15th.

Signed-off-by: Dave Cheney <dave@cheney.net>

### CODE REVIEW

File: release-0.4.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Due to a limitation in Envoy Contour 0.3 exchanged certificate data with Envoy via a shared file system.
+This limitation has been addressed and Contour 0.4 transmits certificate data directly to Envoy in gRPC API response messages. 
+Thanks @sevein
+Fixes #158
+
+### Contour and Envoy can now live in separate pods
+
+Although not currently utilised, Contour and Envoy can now exist in separate pods.
+This is acomplished by the `--xds-address` and `--xds-port` flags passed to both `contour bootstrap` and `contour serve` which can be used to deploy Contour as a ReplicaSet which Envoy is a Daemonset. Thanks to @sevein. Fixes #165
```
Comment (Bradamant3) [inline release-0.4.md]: Rewrite previous two sentences:
"To configure Contour and Envoy to run in separate pods, you can now pass the `--xds-address` and `--xds-port` flags to `contour bootstrap` and `contour serve`. You can then deploy Contour as a ReplicaSet and Envoy as a Daemonset."
(not sure I've quite got it right yet, but the original isn't clear. "Not currently utilised" by what or by whom? Also I'm guessing about the end of that first sentence ...

File: release-0.4.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+### `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation now applies on a per Ingress basis
+
+Prior to Contour 0.4 if _any_ Ingress mentioned a virtual host used the `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation, then all routes for that virtual host would be 301 upgraded.
```
Comment (Bradamant3) [inline release-0.4.md]: "In Contour version 0.3 and earlier, if any Ingress for a virtual host specified the `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation, then all routes for the virtual host were redirected with a 301.

File: release-0.4.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+In Contour 0.4 this behaviour is applied per Ingress, that is, all routes in an Ingress object.
```
Comment (Bradamant3) [inline release-0.4.md]: s/that is,/that is, to

File: release-0.4.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+This permits so-called _split_ Ingress setups where some routes on a virtual host will be 301 ugpraded, and others not.
```
Comment (Bradamant3) [inline release-0.4.md]: This allows a split Ingress setup in which some routes on a virtual host are redirected with a 301, and others are not.

File: release-0.4.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+### `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation now applies on a per Ingress basis
+
+Prior to Contour 0.4 if _any_ Ingress mentioned a virtual host used the `ingress.kubernetes.io/force-ssl-redirect: "true"` annotation, then all routes for that virtual host would be 301 upgraded.
+In Contour 0.4 this behaviour is applied per Ingress, that is, all routes in an Ingress object.
+This permits so-called _split_ Ingress setups where some routes on a virtual host will be 301 ugpraded, and others not. 
+Fixes #250.
+
+### Other bug fixes and improvements in this release
+
+- The Contour docker image no longer bundles the `ocid` and `gcp` authentication plugins as they are not required for `-incluster` deployments.
```
Comment (Bradamant3) [inline release-0.4.md]: s/docker/Docker
s/as/because

File: release-0.4.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+- Daemonset or Deployment examples now specify the `--v2-config-only` flag to instruct Envoy to not fall back to parsing an invalid configuration file as the deprecated v1 JSON format. This should aid debugging `contour bootstrap` issues. Thanks @cmaloney
+- Updated to the latest `envoyproxy/go-control-plane` library. Thanks @vaamarnath. Fixes #225
+- Contour has switched to sirupsen/logrus as its logging library. Fixes #162.
+- Our [troubleshooting][3] documentation now includes a simple way to access Envoy's Admin interface which is useful for examining the state of its route and cluster tables.
+- Various issues related to updating Services and Ingresses in place via `kubectl edit` have been by the introduction of a new caching layer in the translator package.
```
Comment (Bradamant3) [inline release-0.4.md]: Issues related to updating Service and Ingress objects in place with `kubectl edit` are fixed with a new caching layer in the translator package.

**Comment by davecheney** [comment]
Merged, feel free to send a pr.

**Pull request merged**